### PR TITLE
Qualcomm AI Engine Direct - HF LLM Performance and Accuracy Improvement

### PR DESCRIPTION
**Pull request by winskuo-quic**
### Summary
`python examples/qualcomm/oss_scripts/hf_causal_lm.py --prompt "Simply put, the theory of relativity states that" --soc_model SM8750  --device $DEVICE_ID --build_folder build-android/ --decoder_model llama3_2-1b --max_seq_len 128 --ptq 16a8w`

### Test plan
[PLEASE REMOVE] How did you test this PR? Please write down any manual commands you used and note down tests that you have written if applicable.

### CODE REVIEW

File: examples/qualcomm/oss_scripts/llm_utils/decoder_model_wrapper.py [hunk 1]
```diff
@@ -134,32 +191,81 @@ def __init__(self, model):
...
+        QnnPrecomputedRotaryEmbedding). Keeps HF's plumbing intact -- the decoder
+        still calls ``self.rotary_emb(hidden_states, position_ids)`` -- but the
+        wide-dynamic-range inv_freq math now happens on the host in fp32."""
+        decoder = self.model.model
+        rotary_emb = getattr(decoder, "rotary_emb", None)
+        if rotary_emb is None:
+            logging.warning("No rotary_emb found; skipping RoPE precompute.")
+            return
+        # HF API does not have a way to replace, so manually replace it.
+        decoder.rotary_emb = QnnPrecomputedRotaryEmbedding(
```
Comment (winskuo-quic) [inline examples/qualcomm/oss_scripts/llm_utils/decoder_model_wrapper.py:218]: Rope manual override.

File: examples/qualcomm/oss_scripts/llm_utils/decoder_model_wrapper.py [hunk 2]
```diff
@@ -12,12 +12,47 @@
...
+        self.new_values = value_states
+        keys = torch.cat([self.keys, key_states], dim=-2)
+        values = torch.cat([self.values, value_states], dim=-2)
+        return keys, values
+
+    def get_mask_sizes(self, cache_position):
+        return self.max_cache_len, 0
+
+
+class QnnCustomStaticCache(Cache):
```
Comment (winskuo-quic) [inline examples/qualcomm/oss_scripts/llm_utils/decoder_model_wrapper.py:41]: QNN Custom Cache
